### ISSUE TITLE
Failing Test Using async_to_sync in ORM Manager Class

### ISSUE THREAD

**Issue by Palmy523** [django/channels#1286]
I'm currently trying to get a test to pass that involves sending a notification when a create or bulk create is done on a model. I have created a custom Manager that overrides the create and bulk_create methods. 

`from django.db import models

from .tasks import send_notification

class NotificationManager(models.Manager):
    def create(self, *args, **kwargs):
        notification = super(NotificationManager, self).create(*args, **kwargs)
        send_notification(notification)
        return notification

    def bulk_create(self, notifications, **kwargs):
        saved_notifications = super().bulk_create(notifications, **kwargs)
        for notification in saved_notifications:
            send_notification(notification)
        return saved_notifications`

The problem I have, is when I create the Notification object in the test and rely on the task 'send_notification' I get a 'RuntimeError: You cannot use AsyncToSync in the same thread as an async event loop - just await the async function directly.'

The send_notification tasks looks like this right now:

`def send_notification(notification):
	user_channel_id = f'{NOTIFICATION_CONSUMER_NAME}{notification.user.id}'
	data = {
		'type': 'group.message',
		'payload': model_to_dict(notification)
	}
	async_to_sync(channel_layer.group_send)(user_channel_id, data)`

Finally, here is my test

`@pytest.mark.asyncio
@pytest.mark.django_db(transaction=True)
@override_settings(CHANNEL_LAYERS=WEBSOCKET_TEST_SETTINGS)
async def test_push_notification_on_message():
    user = User.objects.create_user(testuser)
    
    consumer_name = f'ws/{NOTIFICATION_CONSUMER_NAME}/'

    communicator1 = WebsocketCommunicator(NotificationConsumer, consumer_name)
    communicator1.scope['user'] = user

    connected, subprotocol = await communicator1.connect()
    channel_layer = get_channel_layer()

    await communicator1.receive_json_from()

    notification = Notification.objects.create(
        user=user,
        link='/test',
        message='test',
        type=Notification.GENERAL
    )

    user_channel_id = f'{NOTIFICATION_CONSUMER_NAME}{notification.user.id}'
    # data = {
    #     'type': 'group.message',
    #     'payload': model_to_dict(notification)
    # }

    # await channel_layer.group_send(user_channel_id, data)
    receivedNotification = await communicator1.receive_json_from()

    assert receivedNotification['payload'] == model_to_dict(notification)

    await communicator1.disconnect()`

If I uncomment the manual send, the test passes as expected, but I want to test that the task is doing the job. 

Any way around this?

**Comment by blueyed** [comment]
Is it only a problem in the test?
Then maybe try without `@pytest.mark.asyncio` / `async dev`, i.e. do not run the test in an event loop.

**Comment by Palmy523** [comment]
Yes only a problem with the test. Running without @pytest.mark.asyncio does not run the test to completion before pytest has finished. I made certain of this by removing the notation and putting an assertion to fail at the end and all tests passed. I'm afraid this isn't going to solve the problem.

**Comment by blueyed** [comment]
I've meant to refactor your test so that it would not need to be async in the first place..
